### PR TITLE
feat: dark mode for plots

### LINKED ISSUE #798: Dark mode for plots
### Is your feature request related to a problem?

While we don't want individual styling options for plots, a nice, consistent dark mode would improve integration into IDEs.

### Desired solution

Add a new, optional, keyword-only parameter `theme` to all plot methods of `ColumnPlotter` and `TablePlotter`.

Type: `Literal["dark", "light"]`
Default value: "light"

* The same plot should have a consistent look for each theme.
* Different plots should have a consistent look for different themes.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by saius02**
Closes #798 

### Summary of Changes

Implemented Dark Mode for Class Tableplotter and Columnplotter.

### CODE REVIEW

File: src/safeds/data/tabular/plotting/_column_plotter.py [hunk 1]
```diff
@@ -54,27 +54,84 @@ def box_plot(self) -> Image:
...
+    def histogram(self, *, max_bin_count: int = 10, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Create a histogram for the values in the column.
 
         Parameters
         ----------
         max_bin_count:
             The maximum number of bins to use in the histogram. Default is 10.
+        theme:
+            The theme for the plot, either "dark" or "light". Default is "light"
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_column_plotter.py:134]:
```suggestion
            The theme for the plot, either "dark" or "light". Default is "light".
```

File: tests/safeds/data/tabular/plotting/test_moving_average_plot.py [hunk 2]
```diff
@@ -59,6 +59,59 @@ def test_should_match_snapshot(
     assert line_plot == snapshot_png_image
 
 
+@pytest.mark.parametrize(
+    ("table", "x_name", "y_name", "window_size"),
+    [
+        (Table({"A": [1, 2, 3], "B": [2, 4, 7]}), "A", "B", 2),
+        # (Table({"A": [1, 1, 2, 2, 3, 3, 4, 4, 5, 5], "B": [2, 4, 6, 8, 10, 12, 14, 16, 18, 20]}), "A", "B", 2),
```
Comment (lars-reimann) [inline tests/safeds/data/tabular/plotting/test_moving_average_plot.py:66]: Still needed? If not, please remove.

File: src/safeds/data/tabular/plotting/_column_plotter.py [hunk 3]
```diff
@@ -87,9 +144,24 @@ def histogram(self, *, max_bin_count: int = 10) -> Image:
...
+                        "xtick.color": "white",
+                        "ytick.color": "white",
+                    },
+                )
+
+            return self._column.to_table().plot.histograms(max_bin_count=max_bin_count)
+
+    def lag_plot(self, lag: int, *, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Create a lag plot for the values in the column.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_column_plotter.py:166]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_column_plotter.py [hunk 4]
```diff
@@ -29,7 +29,7 @@ class ColumnPlotter:
     def __init__(self, column: Column):
         self._column: Column = column
 
-    def box_plot(self) -> Image:
+    def box_plot(self, *, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Create a box plot for the values in the column. This is only possible for numeric columns.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_column_plotter.py:34]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ class TablePlotter:
     def __init__(self, table: Table):
         self._table: Table = table
 
-    def box_plots(self) -> Image:
+    def box_plots(self, *, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Create a box plot for every numerical column.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:40]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 6]
```diff
@@ -61,50 +61,75 @@ def box_plots(self) -> Image:
...
+                            "axes.edgecolor": "white",
+                            "xtick.color": "white",
+                            "ytick.color": "white",
+                        },
+                    )
+                return _figure_to_image(fig)
+
+    def correlation_heatmap(self, *, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Plot a correlation heatmap for all numerical columns of this `Table`.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:134]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 7]
```diff
@@ -122,41 +147,58 @@ def correlation_heatmap(self) -> Image:
...
 
-        return _figure_to_image(fig)
+                plt.tight_layout()
 
-    def histograms(self, *, max_bin_count: int = 10) -> Image:
+            return _figure_to_image(fig)
+
+    def histograms(self, *, max_bin_count: int = 10, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Plot a histogram for every column.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:203]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 8]
```diff
@@ -177,55 +219,80 @@ def histograms(self, *, max_bin_count: int = 10) -> Image:
...
+    def line_plot(
+        self,
+        x_name: str,
+        y_names: list[str],
+        show_confidence_interval: bool = True,
+        *,
+        theme: Literal["dark", "light"] = "light",
+    ) -> Image:
         """
         Create a line plot for two columns in the table.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:297]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 9]
```diff
@@ -177,55 +219,80 @@ def histograms(self, *, max_bin_count: int = 10) -> Image:
...
-        return _figure_to_image(fig)
+            return _figure_to_image(fig)
 
-    def line_plot(self, x_name: str, y_names: list[str], show_confidence_interval: bool = True) -> Image:
+    def line_plot(
+        self,
+        x_name: str,
+        y_names: list[str],
+        show_confidence_interval: bool = True,
+        *,
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:293]:
```suggestion
        *,
        show_confidence_interval: bool = True,
```

Should also be keyword-only, was wrong before.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 10]
```diff
@@ -264,57 +331,70 @@ def line_plot(self, x_name: str, y_names: list[str], show_confidence_interval: b
...
+                rotation=45,
+                horizontalalignment="right",
+            )  # rotate the labels of the x Axis to prevent the chance of overlapping of the labels
+            fig.tight_layout()
+
+            return _figure_to_image(fig)
+
+    def scatter_plot(self, x_name: str, y_names: list[str], *, theme: Literal["dark", "light"] = "light") -> Image:
         """
         Create a scatter plot for two columns in the table.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:400]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 11]
```diff
@@ -352,36 +432,55 @@ def scatter_plot(self, x_name: str, y_names: list[str]) -> Image:
...
+    def moving_average_plot(
+        self,
+        x_name: str,
+        y_name: str,
+        window_size: int,
+        *,
+        theme: Literal["dark", "light"] = "light",
+    ) -> Image:
         """
         Create a moving average plot for the y column and plot it by the x column in the table.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:485]: `theme` parameter should be documented.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 12]
```diff
@@ -122,41 +147,58 @@ def correlation_heatmap(self) -> Image:
         # TODO: implement using matplotlib and polars
         #  https://stackoverflow.com/questions/33282368/plotting-a-2d-heatmap
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:148]: Already done, can be removed.

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 13]
```diff
@@ -122,41 +147,58 @@ def correlation_heatmap(self) -> Image:
...
+                plt.rcParams.update(
+                    {
+                        "text.color": "white",
+                        "axes.labelcolor": "white",
+                        "axes.edgecolor": "white",
+                        "xtick.color": "white",
+                        "ytick.color": "white",
+                    },
+                )
+            import numpy as np
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:163]: Move the import to the top of the function (line 150).

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Comment by TobiasPluecker** [comment]
@lars-reimann is there anything to do for us because there is one change requested

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 14]
```diff
@@ -304,6 +314,8 @@ def line_plot(
             The name(s) of the column(s) to be plotted on the y-axis.
         show_confidence_interval:
             If the confidence interval is shown, per default True.
+        theme:
+            The color theme of the plot. Default is "light".
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:318]: The description for `theme` should be consistent for all functions. The description you used here seems best.

**Review by lars-reimann — CHANGES_REQUESTED** [review]
Should be the last required change.

**Review by lars-reimann — APPROVED** [review]
Great work!

**Pull request merged**